tourcms.js

```diff
@@ -1,799 +1,841 @@
+        return this.request(endpoint, channel)
+    }
+
+    searchVoucher(voucherData = null, channel = 0) {
+
+        if (!voucherData) {
+            voucherData = '<voucher><barcode_data></barcode_data></voucher>'
+        }
+        let postData = this.parseXml(voucherData)
+
+        if (channel == 0) {
+            return (this.request(TourCMS.P_VOUCHER_SEARCH, channel, TourCMS.HTTP_VERB_POST, postData));
+        } else {
+            return (this.request(TourCMS.C_VOUCHER_SEARCH, channel, TourCMS.HTTP_VERB_POST, postData));
+        }
+    }
+
+    // Updating bookings
+
+    updateBooking(bookingData, channel) {
+        return (this.request(TourCMS.C_BOOKING_UPDATE, channel, TourCMS.HTTP_VERB_POST, bookingData));
+    }
+
+    createPayment(paymentData, channel) {
+        return (this.request(TourCMS.C_BOOKING_PAYMENT_NEW, channel, TourCMS.HTTP_VERB_POST, paymentData));
+    }
+
+    logFailedPayment(paymentData, channel) {
+        return (this.request(TourCMS.C_BOOKING_PAYMENT_FAIL, channel, TourCMS.HTTP_VERB_POST, paymentData));
+    }
+
+    spreedlyCreatePayment(paymentData, channel) {
+        return (this.request(TourCMS.C_BOOKING_PAYMENT_SPREEDLY_NEW, channel, TourCMS.HTTP_VERB_POST, paymentData));
+    }
+
+    spreedlyCompletePayment(transactionId, channel) {
+        let endpoint = TourCMS.C_BOOKING_PAYMENT_SPREEDLY_COMPLETE + '?id=' + transactionId
+        return (this.request(endpoint, channel, TourCMS.HTTP_VERB_POST));
+    }
+
+    cancelBooking(channel, bookingId, note) {
+        let xmlString = "<booking><booking_id>" + bookingId + "</booking_id><note>" + note + "</note></booking>";
+        let postData = this.parseXml(xmlString);
+        return this.request(TourCMS.C_BOOKING_CANCEL, channel, TourCMS.HTTP_VERB_POST, postData);
+    }
+
+    deleteBooking(booking, channel) {
+        let endpoint = TourCMS.C_BOOKING_DELETE + '?booking_id=' + booking
+        return (this.request(endpoint, channel, TourCMS.HTTP_VERB_POST));
+    }
+
+    checkOptionAvailability(booking, tourComponentId, channel) {
+        let endpoint = TourCMS.C_BOOKING_OPTIONS_CHECKAVAIL + '?booking_id=' + booking + '&tour_component_id=' + tourComponentId
+        return (this.request(endpoint , channel));
+    }
+
+    bookingAddComponent(componentData, channel) {
+        return (this.request(TourCMS.C_BOOKING_COMPONENT_NEW, channel, TourCMS.HTTP_VERB_POST, componentData));
+    }
+
+    bookingUpdateComponent(componentData, channel) {
+        return (this.request(TourCMS.C_BOOKING_COMPONENT_UPDATE, channel, TourCMS.HTTP_VERB_POST, componentData));
+    }
+
+    bookingRemoveComponent(componentData, channel) {
+        return (this.request(TourCMS.C_BOOKING_COMPONENT_DELETE, channel, TourCMS.HTTP_VERB_POST, componentData));
+    }
+
+    addNoteToBooking(booking, channel, text, type) {
+        let xmlString = "<booking><booking_id>"+booking+"</booking_id><note><text>"+text+"</text><type>"+type+"</type></note></booking>";
```

Comment:
Better to leave the user to generate the XML  or being sure that we have all possible required fields

Reply:
done